source-code-git/fs/src/implementations/memoryFs.ts

```diff
@@ -157,15 +157,17 @@ export function createMemoryFs(): NodeishFilesystem {
 			path: Parameters<NodeishFilesystem["symlink"]>[1],
 		) {
 			path = normalPath(path)
+			target = target.startsWith("/") ? target : `${path}/../${target}`
```

Comment:
If we force absolute paths, wouldn't it make sense to use `assertIsAbsolutePath` and explicitly provide developers with a clear error message instead of implicitly fixing the path (that might lead to confusion because relative paths can sometimes be used and sometimes not)? 

Like in https://github.com/inlang/inlang/blob/3cc7b69936b393dfa6e3fc13a22b39e7734944ed/source-code-git/fs/src/utilities/fromJson.ts#L14

Reply:
@araknast Are you confident that forcing absolute paths is the right decision?

EDIT: Absolute paths are less ambiguous for sure and cwd doesn't exist in the browser. Seems to me that trading off "works in every environment" for slight DX improvements for node (by not requiring absolute paths in node) is a good one.

Reply:
> If we force absolute paths, wouldn't it make sense to use assertIsAbsolutePath and explicitly provide developers with a clear error message instead of implicitly fixing the path (that might lead to confusion because relative paths can sometimes be used and sometimes not)?

Ideally we would be able to use `assertIsAbsolutePath` to ensure all paths begin with `/` but I don't think it would be worth the trouble of making it work with isomorphic-git. There is also the issue of `mockEnvironment`, and the reason this issue exists in the first place in that symlinks on 'real' filesystems often use relative paths. Using `assertIsAbsolutePath` would mean that we could not create a mock environment with a directory that contains a relative symbolic link (the reason why the error in #753 was occurring.

> Are you confident that forcing absolute paths is the right decision?
>
> EDIT: Absolute paths are less ambiguous for sure and cwd doesn't exist in the browser. Seems to me that trading off "works in every environment" for slight DX improvements for node (by not requiring absolute paths in node) is a good one.

All paths in memoryFs are 'absolute' anyways since they are always relative to the root of the filesystem. Internally a '/' is prepended to the path as part of the `normalPath` function. The behavior is identical to node as if `cwd` was set to `/`. 

To be clear, I don't necessarily want to enforce use of absolute paths in the api (for the reasons outlined above), only to ensure all paths that are passed around internally in memoryFs are absolute. This was the intended behavior before, and this patch simply extends it to the `symlink` command.

Reply:
@araknast Got it, makes sense!